src/ui/component/copyToClipboard/copyToClipboard.tsx

```diff
@@ -10,23 +12,45 @@ type CopyToClipBoardProps = {
 
 export const CopyToClipboard = ({ onCopied, textToCopy }: CopyToClipBoardProps): JSX.Element => {
   const theme = useAppStore(state => state.theme)
+  const [iconName, setIconName] = useState<IconName>(`copy-${theme}`)
```

Comment:
Restrict the type of the icons of the state, you just want a few subset of the `IconNames`. So specify the union types explicitly.